src/backend/joanie/core/filters/admin.py

```diff
@@ -70,6 +74,43 @@ def filter_by_query(self, queryset, _name, value):
         """
         return queryset.filter(translations__title__icontains=value).distinct()
 
+    def filter_by_state(self, queryset, _name, value):
+        """
+        Filter resource by looking for states that match the one provided
+        in the "state" parameter
+        """
+        now = django_timezone.now()
+        if value == models.CourseState.ONGOING_OPEN:
+            queryset = queryset.filter(
+                start__lt=now, end__gt=now, enrollment_end__gt=now
+            )
+        elif value == models.CourseState.FUTURE_OPEN:
+            queryset = queryset.filter(start__gte=now, enrollment_end__gt=now)
+        elif value == models.CourseState.ARCHIVED_OPEN:
+            queryset = queryset.filter(
+                start__lt=now,
+                end__lte=now,
+                enrollment_start__lt=now,
+                enrollment_end__gt=now,
+            )
+        elif value == models.CourseState.FUTURE_CLOSED:
+            queryset = queryset.filter(start__gte=now, enrollment_end__lte=now)
+        elif value == models.CourseState.ONGOING_CLOSED:
+            queryset = queryset.filter(
+                start__lte=now, end__gt=now, enrollment_end__lte=now
```

Comment:
ongoing doesn't care about enrollment date.

Reply:
The code in compute state is:
```
if end > now:
    if enrollment_end > now:
        # ongoing open
        return CourseState(CourseState.ONGOING_OPEN, enrollment_end)
    # ongoing closed
    return CourseState(CourseState.ONGOING_CLOSED)
```
The enrollment end_date is the deciding factor between ONGOING_OPEN and ONGOING_CLOSED